### PR TITLE
(SIMP-3633) Support for local system builds

### PR DESCRIPTION
**Pull request by trevor-vaughan**
* Removed all 'mock' support
* Bound the build to only the OS upon which you are building
* Removed legacy support code
* Designed for use within a Docker environment
  * Technically, you can install and run it inside mock but it's not
    going to do any of the heavy lifting for you any longer
* Is *NOT* backwards compatible with anything for the most part

SIMP-3701 #close

### CODE REVIEW

**Comment by jeefberkey** [comment]
Rebase required

**Comment by jeannegreulich** [comment]
When I tried editing running the RPM build on centos 6 by uncommenting the centos 6 portion in the nodeset/default.yaml file under rpm_docker, it fails.  Says the default.yaml is not valid.   I did not trouble shoot beyond that.

**Comment by trevor-vaughan** [comment]
@jeannegreulich I've fixed the indentation in the nodeset and run through three different runs of builds. The EL6 build is disabled in the nodeset by default for now since I don't have the supporting code for just building the `arch` RPMs yet. That's next sprint.

**Comment by trevor-vaughan** [comment]
Also, this code was fine. `simp-core` had the issues.

File: lib/simp/rake/build/pkg.rb [hunk 1]
```diff
@@ -728,63 +638,168 @@ def define_tasks
...
+
+            new_rpm = Simp::Rake::Pkg.new(Dir.pwd, opts[:unique_namespace], @simp_version)
+
+            new_rpm_info = Simp::RPM.new(new_rpm.spec_file)
+
+            if opts[:check_git]
+              require_tag = false
+
+              # We remove any leading 'v' chars since some projects use them
+              latest_tag = %x(git describe --abbrev=0 --tags 2>/dev/null).strip.gsub(/^v/,'')
```
Comment (lnemsick-simp) [inline lib/simp/rake/build/pkg.rb]: This does not work for our forked modules that have tags that begin with simp-, simp6.0.0-, or have suffixes.  So, these modules get erroneously rebuilt.

File: lib/simp/rpm.rb [hunk 2]
```diff
@@ -27,15 +28,58 @@ def self.sh(args)
...
+        ver1 = Gem::Version.new(File.basename(rpm1, '.rpm').split('-')[-1..-1].join('-'))
+        ver2 = Gem::Version.new(File.basename(rpm2, '.rpm').split('-')[-1..-1].join('-'))
+
+        if ver1 > ver2
+          return ver1
+        else
+          return ver2
+        end
+      rescue ArgumentError, NoMethodError
+        fail("Could not compare RPMs '#{rpm1}' and '#{rpm2}'")
```
Comment (lnemsick-simp) [inline lib/simp/rpm.rb]: This fails for simp RPM when I called pkg:check_published target.  Output is:
Could not compare RPMs 'simp-6.1.0-RC1.el7.noarch.rpm' and 'simp-6.0.0-1.el7.noarch.rpm'.
Had to replace the fail with a $stderr.puts in order to see what pkg:check_published did with modules.

File: lib/simp/rpm.rb [hunk 3]
```diff
@@ -27,15 +28,58 @@ def self.sh(args)
...
+
+    # Returns whether or not the first RPM is newer than the second RPM
+    def self.newer?(rpm1, rpm2)
+      return Simp::RPM.newer(rpm1, rpm2) == rpm1
+    end
+
+    # Returns the name of the newer RPM
+    #
+    # Will simply return the second RPM name if the two are equal
+    def self.newer(rpm1, rpm2)
```
Comment (lnemsick-simp) [inline lib/simp/rpm.rb]: This does not work as expected.  It tells me the remote is newer when the local is newer.  For example, with  my own debugging in, got results such as the following:put
Found newer remote RPM for simp-adapter
published="simp-adapter-0.0.3-0.el7.noarch.rpm"
new=#<Simp::RPM:0x00000003e51688 @basename="simp-adapter", @version="0.0.4", @release="0.el7", @full_version="0.0.4-0.el7", @name="simp-adapter-0.0.4-0
.el7", @arch="noarch", @package_name="simp-adapter-0.0.4-0.el7.noarch.rpm", @verbose=false, @packages=["simp-adapter", "simp-adapter-pe"], @signature=n
il>

**Comment by trevor-vaughan** [comment]
This also closes SIMP-1437 and SIMP-4792

**Pull request merged**